datafusion/physical-expr/src/aggregate/approx_median.rs

```diff
@@ -67,7 +67,11 @@ impl AggregateExpr for ApproxMedian {
     }
 
     fn create_accumulator(&self) -> Result<Box<dyn Accumulator>> {
-        self.approx_percentile.create_accumulator()
+        self.approx_percentile.create_accumulator().map_err(|_| {
+            DataFusionError::Execution(
+                "approx median requires at least one non-null value".to_string(),
+            )
+        })
```

Comment:
I'm not sure if this wrapping will prevent having an empty input? Could you add some tests to check this?